rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/InterpretedLanguageContributions.java

```diff
@@ -234,16 +228,51 @@ private static CompletableFuture<Boolean> nonNull(CompletableFuture<?> x) {
         return x.thenApply(Objects::nonNull);
     }
 
-    private static CompletableFuture<Boolean> summaryConfigLookup(CompletableFuture<IConstructor> conf, String parameter) {
-        return conf.thenApply( d -> {
-            if (d == null) {
-                return false;
+    private static CompletableFuture<SummaryConfig> scheduledSummaryConfig(CompletableFuture<ISet> contributions, String summarizer) {
+        return contributions.thenApply(c -> {
+            var constructor = getContribution(c, summarizer);
+            if (constructor != null) {
+                return new SummaryConfig(
+                    isTrue(constructor, "providesDocumentation"),
+                    isTrue(constructor, "providesDefinitions"),
+                    isTrue(constructor, "providesReferences"),
+                    isTrue(constructor, "providesImplementations"));
+            } else {
+                return SummaryConfig.FALSY;
             }
-            var val = d.asWithKeywordParameters().getParameter(parameter);
-            return !(val instanceof IBool) || ((IBool)val).getValue();
         });
     }
 
+    private static CompletableFuture<SummaryConfig> ondemandSummaryConfig(CompletableFuture<ISet> contributions) {
+        return contributions.thenApply(c ->
+            new SummaryConfig(
+                hasContribution(c, DOCUMENTER_NAME),
+                hasContribution(c, DEFINER_NAME),
+                hasContribution(c, REFERRER_NAME),
+                hasContribution(c, IMPLEMENTER_NAME)));
+    }
+
+    private static @Nullable IConstructor getContribution(ISet contributions, String name) {
+        return contributions
+            .stream()
+            .map(IConstructor.class::cast)
+            .filter(cons -> cons.getConstructorType().getName().equals(name))
+            .findAny()
+            .orElse(null);
+    }
+
+    private static boolean hasContribution(ISet contributions, String name) {
+        return getContribution(contributions, name) != null;
+    }
+
+    private static boolean isTrue(@Nullable IConstructor constructor, String parameter) {
+        if (constructor == null) {
+            return false;
+        }
+        var val = constructor.asWithKeywordParameters().getParameter(parameter);
```

Comment:
needs null check for robustness sake (NPE if the parameter is not set on the constructor)

Reply:
doesn't it do that? the next line is: `(val instanceof IBool)` , that returns `false` for `null` right?

Reply:
Yes:

```
jshell> null instanceof String
$1 ==> false
```

I didn't know that! Never to old to  learn :-)

Reply:
Fun fact, you can cast null to anything, but instanceof null is always false.

Quite convient.